app/src/screens/PersonCredentialLoading.tsx

```diff
@@ -0,0 +1,258 @@
+import { useAgent, useCredentialByState, useProofByState } from '@credo-ts/react-hooks'
+import {
+  BifoldAgent,
+  Button,
+  ButtonType,
+  Screens,
+  Stacks,
+  TOKENS,
+  testIdWithKey,
+  useContainer,
+  useStore,
+  useTheme,
+  EventTypes as BifoldEventTypes,
+  BifoldError,
+} from '@hyperledger/aries-bifold-core'
+import { useNavigation } from '@react-navigation/native'
+import React, { useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import {
+  DeviceEventEmitter,
+  EmitterSubscription,
+  Modal,
+  SafeAreaView,
+  ScrollView,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native'
+
+import PersonCredentialSpinner from '../components/PersonCredentialSpinner'
+import { connectToIASAgent, authenticateWithServiceCard, WellKnownAgentDetails } from '../helpers/BCIDHelper'
+import { useAttestation } from '../hooks/useAttestation'
+import { AttestationEventTypes } from '../services/attestation'
+import { BCState } from '../store'
+
+export default function PersonCredentialLoading() {
+  const { ColorPallet, TextTheme } = useTheme()
+  const [store] = useStore<BCState>()
+  const [remoteAgentDetails, setRemoteAgentDetails] = useState<WellKnownAgentDetails | undefined>()
+  const receivedProofRequests = useProofByState(ProofState.RequestReceived)
+  const timer = useRef<NodeJS.Timeout>()
+  const logger = useContainer().resolve(TOKENS.UTIL_LOGGER)
+  const receivedCredentialOffers = useCredentialByState(CredentialState.OfferReceived)
+  const { loading: attestationLoading } = useAttestation()
+  const { agent } = useAgent()
+  if (!agent) {
+    throw new Error('Unable to fetch agent from Credo')
+  }
+  const navigation = useNavigation()
+  const { t } = useTranslation()
+  const attestationProofRequestWaitTimeout = 10000
+  const [didCompleteAttestationProofRequest, setDidCompleteAttestationProofRequest] = useState<boolean>(false)
+
+  const styles = StyleSheet.create({
+    container: {
+      height: '100%',
+      backgroundColor: ColorPallet.brand.modalPrimaryBackground,
+      padding: 20,
+    },
+    image: {
+      marginTop: 80,
+    },
+    messageContainer: {
+      alignItems: 'center',
+      marginTop: 40,
+    },
+    messageText: {
+      fontWeight: TextTheme.normal.fontWeight,
+      textAlign: 'center',
+      marginTop: 30,
+    },
+    controlsContainer: {
+      marginTop: 'auto',
+      margin: 20,
+    },
+    delayMessageText: {
+      textAlign: 'center',
+      marginTop: 20,
+    },
+  })
+
+  useEffect(() => {
+    const connect = async () => {
+      try {
+        const remoteAgentDetails = await connectToIASAgent(agent, store, t)
+        setRemoteAgentDetails(remoteAgentDetails)
+        logger.error(`Connected to IAS agent, connectionId: ${remoteAgentDetails.connectionId}`)
+      } catch (err) {
+        logger.error(`Failed to connect to IAS agent, error: ${(err as BifoldError).message}`)
+      }
+    }
+    connect()
+  }, [])
```

Comment:
I think the rule of thumb, any variables used inside the `useEffect` hook should be included in the dependencies array ([`agent`, `store` and `t`]), even if you don't care if they change. This ensures that the effect is aware of all the variables it relies on and can react appropriately when they change.

From the docs: "One important principle in React is local reasoning: the ability to understand what a component or hook does by looking at its code in isolation. Hooks should be treated like “black boxes” when they are called. For example, a custom hook might have used its arguments as dependencies to memoize values inside it:"

https://react.dev/reference/rules/components-and-hooks-must-be-pure

Maybe this would work around the issue?

```
  const connect = async () => {
    try {
      const remoteAgentDetails = await connectToIASAgent(agent, store, t);
      setRemoteAgentDetails(remoteAgentDetails);
      logger.error(`Connected to IAS agent, connectionId: ${remoteAgentDetails.connectionId}`);
    } catch (err) {
      logger.error(`Failed to connect to IAS agent, error: ${(err as BifoldError).message}`);
    }
  };

  useEffect(() => {
    connect();
  }, []); // Empty dependencies array since connect function is independent of props or state
```